packages/@aws-cdk/alexa-ask/package.json

```diff
@@ -50,11 +50,15 @@
     "cfn2ts": "cfn2ts",
     "build+test+package": "npm run build+test && npm run package",
     "build+test": "npm run build && npm test",
-    "compat": "cdk-compat"
+    "compat": "cdk-compat",
+    "gen": "cfn2ts"
   },
   "cdk-build": {
     "cloudformation": "Alexa::ASK",
     "jest": true,
+    "pre": [
```

Comment:
This feels unnecessary. We can just code knowledge of the `gen` target into `cdk-build`, and get rid of `cdk-build.pre` altogether.

(Yes, that means moving everything that's currently in `cdk-build.pre` into `scripts.gen`. If you don't want to that, I'm okay with leaving `cdk-build.pre` in, but I'd still prefer that the *tool* knows about `gen`ning so we don't have to repeat this in every package file).

Reply:
By "prefer the tool knows about `gen`ning" you mean have `cdk-build` call `scripts.gen` instead of the `pre`?

Reply:
Good call. I removed `npm run gen` from all  `pre` directive. `cdk-build` will now execute `gen` unless `--skip-gen` was provided. 

I moved all code generation logic to `gen`, this include code generation commands in non cfn modules as well. I didn't remove `pre` completely since there are cases in which it is used for things like cleaning folders, which are not code generation related.